src/Psalm/Internal/Provider/ReturnTypeProvider/GetObjectVarsReturnTypeProvider.php

```diff
@@ -126,7 +156,12 @@ public static function getGetObjectVarsReturnType(
                 return new TKeyedArray(
                     $properties,
                     null,
-                    $class_storage->final ? null : [Type::getString(), Type::getMixed()],
+                    $class_storage->final
+                        || $class_storage->name === UnitEnum::class
+                        || $class_storage->name === BackedEnum::class
```

Comment:
Perhaps update `classImplements()` to work with interfaces?

Reply:
I'd suggest using `classImplements() || interfaceExtends()`. Extension of enum interfaces is valid in PHP: https://3v4l.org/eKJ49

This will also cover the case of `BackedEnum` (as it extends `UnitEnum`). And to save some CPU cycles you can check `$class_storage->is_enum` before checking the inheritance tree.

Reply:
The test case for that may look something like this;
```php
interface I extends BackedEnum {}

/** @return array{name: non-empty-string, value: int|string} */
function f(I $e): void {
   return get_object_vars($e);
}
```

Reply:
Another thought, maybe we should mark all enums as final? It is practically true, since enums cannot have properties.

Reply:
It would match the way enums are represented in PHP internally:
![image](https://user-images.githubusercontent.com/57403/209022606-73286d7f-e120-4633-9a72-a18b37467dc7.png)

so it sounds good to me. We'll still need to check `interfaceExtends()`, as we may be dealing with an interface extending one of the enum interfaces, and that interface won't be final itself.